Core/source/core/pgp-key.ts

```diff
@@ -311,12 +316,13 @@ export class PgpKey {
     return { original: armored, normalized, keys: await Promise.all(keys.map(PgpKey.details)) };
   }
 
-  public static details = async (k: OpenPGP.key.Key): Promise<KeyDetails> => {
+  public static details = async (k: Key): Promise<KeyDetails> => {
     const keys = k.getKeys();
-    const algoInfo = k.primaryKey.getAlgorithmInfo();
-    const algo = { algorithm: algoInfo.algorithm, bits: algoInfo.bits, curve: (algoInfo as any).curve, algorithmId: openpgp.enums.publicKey[algoInfo.algorithm] };
-    const created = k.primaryKey.created.getTime() / 1000;
-    const exp = await k.getExpirationTime('encrypt');
+    const algoInfo = k.keyPacket.getAlgorithmInfo();
+    const algo = { algorithm: algoInfo.algorithm, bits: algoInfo.bits, curve: (algoInfo as any).curve,
+      algorithmId: enums.publicKey[algoInfo.algorithm] };
+    const created = k.keyPacket.created.getTime() / 1000;
+    const exp = await k.getExpirationTime();
```

Comment:
Similarly expiration here - to track until discussed

Reply:
@tomholub I've backported old function from openpgp v4

Reply:
ok for now